Peer doesn't recognise absolute windows paths for snapshot root dir

Peer fails to start when a valid windows path is provided for the snapshot root dir
```
panic: Error in instantiating ledger provider: invalid path: C:\_SourceCode\github-mine\FabricOnWindows\test-env\data\peer-org1\data\snapshots. The path for the snapshot dir is expected to be an absolute path
```

Looking at the source code in fabric 
```
	if !path.IsAbs(snapshotsRootDir) {
		return errors.Errorf("invalid path: %s. The path for the snapshot dir is expected to be an absolute path", snapshotsRootDir)
	}
```

But in go, path.IsAbs is implemented as
```
// IsAbs reports whether the path is absolute.
func IsAbs(path string) bool {
        return len(path) > 0 && path[0] == '/'
}
```

there is an alternative implementation in path/filepath which has platform specific implementations which should be moved to. 
(All other IsAbs in the fabric codebase use filepath)

--- a/core/ledger/kvledger/kv_ledger_provider.go
+++ b/core/ledger/kvledger/kv_ledger_provider.go
@@ -10,7 +10,7 @@ import (
 	"bytes"
 	"fmt"
 	"os"
-	"path"
+	"path/filepath"
 
 	"github.com/golang/protobuf/proto"
 	"github.com/hyperledger/fabric-protos-go/common"
@@ -240,7 +240,7 @@ func (p *Provider) initLedgerStatistics() {
 
 func (p *Provider) initSnapshotDir() error {
 	snapshotsRootDir := p.initializer.Config.SnapshotsConfig.RootDir
-	if !path.IsAbs(snapshotsRootDir) {
+	if !filepath.IsAbs(snapshotsRootDir) {
 		return errors.Errorf("invalid path: %s. The path for the snapshot dir is expected to be an absolute path", snapshotsRootDir)
 	}
 

